lib/collection.coffee

```diff
@@ -0,0 +1,95 @@
+events    = require 'events'
+inherits_ = require 'inherits-underscore'
+Model     = require './model'
+
+module.exports =
+
+class Collection extends events.EventEmitter
+
+  inherits_ Collection::
+
+  constructor: (name, models) ->
+
+    throw new Error 'missing name'  unless name
+
+    @name   = name
+    @models = []
+
+    [].concat(models).filter(Boolean).forEach (x) =>
+      @add x
```

Comment:
why do you need to clone the array here?

and if cloning is necessary `Array::slice` looks cleaner to me.

`models.slice().filter(Boolean).forEach @add.bind this`